mods/ascent/mobs/bodyparts_insectoid.dm

```diff
@@ -77,4 +77,31 @@
 
 /obj/item/organ/internal/eyes/insectoid
 	name = "compound ocelli"
-	icon_state = "eyes-compound"
\ No newline at end of file
+	icon_state = "eyes-compound"
+
+/obj/item/organ/internal/egg_sac/insectoid
+	name = "gyne egg-sac"
+	action_button_name = "Produce Egg"
+	var/egg_metabolic_cost = 100
+
+/obj/item/organ/internal/egg_sac/insectoid/refresh_action_button()
+	. = ..()
+	if(.)
+		action.button_icon_state = "serpentid-threat"
+		if(action.button) action.button.UpdateIcon()
+
+/obj/item/organ/internal/egg_sac/insectoid/attack_self(var/mob/user)
+	. = ..()
+	var/mob/living/carbon/H = user
+	if(.)
+		if(H.incapacitated())
+			to_chat(H, SPAN_WARNING("You can't produce eggs in your current state."))
+			return		
+		if(H.nutrition < egg_metabolic_cost)
+			to_chat(H, SPAN_WARNING("You are too ravenously hungry to produce more eggs."))
+			return
+		if(do_after(H, 5 SECONDS, H, FALSE))
+			H.adjust_nutrition(-1 * egg_metabolic_cost)
+			H.visible_message(SPAN_NOTICE("\icon[H] [H] produces a fresh, new egg."))
```

Comment:
`fresh, new egg` is a bit redundant isn't it?
Could also do something like `[H] carelessly deposits an egg on \the [loc].`

Reply:
**carelessly** it's gold. I'm in.